sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -2838,9 +2850,14 @@ private List<Document> flushBatch() throws IOException, InterruptedException {
           responseEntity = new BufferedHttpEntity(response.getEntity());
         } catch (java.io.IOException ex) {
           if (spec.getRetryConfiguration() == null || !isRetryableClientException(ex)) {
-            throw ex;
+            if (spec.getThrowWriteErrors()) {
+              throw ex;
+            } else {
+              elasticResponseExceptionMessage = ex.getMessage();
+            }
+          } else {
+            LOG.error("Caught ES timeout, retrying", ex);
           }
-          LOG.error("Caught ES timeout, retrying", ex);
         }
 
         if (spec.getRetryConfiguration() != null
```

Comment:
I believe we should also check that `elasticResponseExceptionMessage != null` as part of this predicate, because we don't need to go through the retry flow in the case where we've already verified that the exception is not retryable above.